redditbot.py

```diff
@@ -30,47 +32,45 @@ def save_cache(cache):
 		f.write(json.dumps(cache, indent="\t"))
 
 def create_reddit_response(match_info):
-	# for hero_match in match_info["heroes"]:
+	# for hero_match in match_info.heroes:
 	# 	print(hero_match)
-	response = f"Looks like this is match {match_info['match_id']}, which started {match_info['minutes_diff']} minutes before the clip was taken."
-	response += f"\n\nhttps://www.opendota.com/matches/{match_info['match_id']}"
+	response = f"Looks like this is match {match_info.match}, which started {match_info.minutes_diff} minutes before the clip was taken."
+	response += f"\n\nhttps://www.opendota.com/matches/{match_info.id}"
 	response += reddit_comment_footer
 	return response
 
-def bot_check_posts():
+def bot_check_posts(reddit):
 	cache = read_cache()
 	for post in reddit.subreddit("dota2").new(limit=100):
 		if post.id in cache["replied_posts"]:
 			continue # already replied to this post
 		match = re.match(r"^https?://clips\.twitch\.tv/([^\?]*)(\?.*)?$", post.url)
 		if match:
 			slug = match.group(1)
-			match_info = None
-			try: 
+			try:
 				match_info = finder.find_match(slug)
 			except finder.ClipFinderException as e:
-				print(f"encountered {type(e).__name__} for slug {slug} on post {post.id}")
+				print(f"encountered {e} for slug {slug} on post {post.id}")
 				pass
-			if match_info is not None:
```

Comment:
This actually needs to keep the if statement. If it is taken out like this, it will try to use a null match_info here